### PR TITLE
Update couchbase API to 3.0.0

### PR DESCRIPTION
**Pull request by mchataigner**
This is a tentative to update couchbase code to 3.0.0 API.
I will need a Couchbase specialist to validate this code.

### CODE REVIEW

**Comment by auvipy** [comment]
python 3.7 n 3.8 is passing the remaining r failing. can you check?

**Comment by auvipy** [comment]
can you change the Travis Linux from focal to bionic to python 2.7, 3.5, and 3.6?

**Comment by mchataigner** [comment]
done in a separate PR https://github.com/celery/celery/pull/6096

**Comment by mchataigner** [comment]
It seems couchbase python client 3.0.0+ has a requirement for python 3.7 on Windows.
According to their readme:
`These are available for at least Python 3.7 on Windows, but we will endeavour to add more`

**Comment by auvipy** [comment]
lelts keep this for future version of celery

**Comment by auvipy** [comment]
this should be rebased on master for celery v5

File: celery/backends/couchbase.py [hunk 1]
```diff
@@ -16,12 +16,15 @@
...
+    from couchbase.cluster import Cluster, ClusterOptions
+    from couchbase.auth import PasswordAuthenticator
+    # from couchbase.connection import Connection
+    # from couchbase.exceptions import NotFoundError
+    from couchbase_core._libcouchbase import FMT_AUTO
+        # from couchbase import FMT_AUTO
 except ImportError:
-    Couchbase = Connection = NotFoundError = None   # noqa
+    Cluster = PasswordAuthenticator = ClusterOptions =  None
+    # Couchbase = Connection = NotFoundError = None   # noqa
```
Comment (thedrow) [inline celery/backends/couchbase.py:27]: Please do not leave commented out code in the codebase.

**Review by thedrow — CHANGES_REQUESTED** [review]

**Comment by thedrow** [comment]
Unfortunately you didn't allow edits from maintainers.
I'm going to replace this PR as I've done the necessary work to merge this.

**Pull request closed without merging**